src/app/send/add-edit.component.ts

```diff
@@ -26,11 +26,21 @@ export class AddEditComponent extends BaseAddEditComponent {
     }
 
     async refresh() {
+        this.password = null;
         const send = await this.loadSend();
         this.send = await send.decrypt();
         this.hasPassword = this.send.password != null && this.send.password.trim() !== '';
         this.deletionDate = this.dateToString(this.send.deletionDate);
         this.expirationDate = this.dateToString(this.send.expirationDate);
-        this.password = null;
+    }
+
+    cancel() {
+        this.onCancelled.emit(this.send);
```

Comment:
Do you need to check for `ths.onCancelled != null`? Not sure, maybe not, but if the delegate method isn't set could throw an exception here.

Reply:
onCancelled has a default emitter [here](https://github.com/bitwarden/jslib/blob/8a3b551c6e4113035dc40a379f43f8778d208ffe/src/angular/components/send/add-edit.component.ts#L34), and that's all that this call uses.